### DISCUSSION TITLE
Initialization (loading the main module)

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#148] [Ideas]
_Based on issue #145_

Currently, the python initialization method (`Py_InitializeFromConfig`) depends on the environment (global or venv). In case of global environment, the standard initialization (`PyConfig_InitPythonConfig`) is performed, while in case of venv, the isolated initialization (`PyConfig_InitIsolatedConfig`) is performed. I know we discussed this in #1, but I don't understand what is the reasoning behind using isolated configuration in venv. In any case, it has to be determined whether the option to choose isolated or non-isolated configuration will still be available, and if so, how can it be determined (environment variables?).

Theoretically, an isolated setup is recommended for embedded Python, but I think the non-isolated version makes more sense for this project. The isolated one ignores any PYTHON* environment variables, e.g. **PYTHONPATH** and PYTHONHOME. Of course `PYTHONPATH` can be processed in `pyrx_onload.py` file, but the question arises whether there is a need to reinvent the wheel.

**Comment by CEXT-Dan** [comment]
I don’t remember, I will come up with a better method of letting the user setup PyPreConfig and PyConfig, something installing or updating won’t overwrite

**Reply by CEXT-Dan** [reply]
It seems the reason why is because, VENV does not seem to work with Py_InitializeFromConfig
even if I set the python.exe path to venv\Scripts

**Reply by CEXT-Dan** [reply]
actually, I need to test this more, it may work
